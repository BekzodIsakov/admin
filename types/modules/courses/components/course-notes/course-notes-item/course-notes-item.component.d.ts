export const CourseNotesItem: any;
//# sourceMappingURL=course-notes-item.component.d.ts.map