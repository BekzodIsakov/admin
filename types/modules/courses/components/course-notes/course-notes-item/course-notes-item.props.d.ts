export function useCourseNotesItemProps({ lessonId, chapterId }: {
    lessonId: any;
    chapterId: any;
}): {
    deleteNote: (id: any) => void;
    editNote: (e: any) => void;
};
//# sourceMappingURL=course-notes-item.props.d.ts.map