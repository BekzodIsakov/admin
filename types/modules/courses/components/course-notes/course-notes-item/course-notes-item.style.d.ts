export const CourseNotesItemWrapper: any;
export const CourseNotesItemTime: any;
export const CourseNotesCenter: any;
export const CourseNotesCenterTop: any;
export const CourseNotesItemTitle: any;
export const CourseNotesItemSubtitle: any;
export const CourseNotesItemText: any;
export const CourseNotesItemRight: any;
export const CourseNotesItemBtn: any;
//# sourceMappingURL=course-notes-item.style.d.ts.map