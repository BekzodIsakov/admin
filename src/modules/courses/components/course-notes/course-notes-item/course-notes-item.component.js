import React from 'react';
import { hoc } from '@utils/hoc';
import { SvgIcon } from '@components/atoms/index';
import { useCourseNotesItemProps } from './course-notes-item.props';

import {
  CourseNotesItemWrapper,
  CourseNotesItemTime,
  CourseNotesCenter,
  CourseNotesCenterTop,
  CourseNotesItemTitle,
  CourseNotesItemSubtitle,
  CourseNotesItemText,
  CourseNotesItemRight,
  CourseNotesItemBtn
} from './course-notes-item.style';

const CourseNotesItem = hoc(
  useCourseNotesItemProps,
  ({ time, title, subtitle, text, id, deleteNote, editNote, onTimeClick }) => {
    return (
      <CourseNotesItemWrapper>
        <CourseNotesItemTime
          data-testid='course-notes-timeBtn'
          onClick={onTimeClick}
        >
          {time}
        </CourseNotesItemTime>

        <CourseNotesCenter>
          <CourseNotesCenterTop>
            <div>
              <CourseNotesItemTitle>{title}</CourseNotesItemTitle>
              <CourseNotesItemSubtitle data-testid='course-notes-subtitle'>
                {' '}
                {subtitle}{' '}
              </CourseNotesItemSubtitle>
            </div>
            <CourseNotesItemRight>
              <CourseNotesItemBtn
                data-testid={`course-notes-edit-${id}`}
                onClick={editNote.bind(null, { id, text })}
              >
                <SvgIcon fill='#000' size='14px' iconName='edit-icon' />
              </CourseNotesItemBtn>
              <CourseNotesItemBtn
                data-testid='course-notes-delete'
                onClick={deleteNote.bind(null, id)}
              >
                <SvgIcon fill='#000' size='14px' iconName='delete-icon' />
              </CourseNotesItemBtn>
            </CourseNotesItemRight>
          </CourseNotesCenterTop>
          <CourseNotesItemText data-testid='course-notes-text'>
            {text}
          </CourseNotesItemText>
        </CourseNotesCenter>
      </CourseNotesItemWrapper>
    );
  }
);

export { CourseNotesItem };
