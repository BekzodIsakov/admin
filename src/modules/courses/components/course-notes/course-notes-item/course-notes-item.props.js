import { useDispatch } from 'react-redux';
import { coursesActions, useActiveLessonSelector } from '@courses/store';

export const useCourseNotesItemProps = ({ lessonId, chapterId }) => {
  const lesson = useActiveLessonSelector();
  const dispatch = useDispatch();

  const editNote = e => {
    dispatch(
      coursesActions.editNotes({
        available: true,
        text: e.text,
        id: e.id,
        lessonId: lessonId,
        chapterId: chapterId
      })
    );
  };

  const deleteNote = id => {
    const payload = {
      course_id: lesson?.course_id,
      chapter_id: chapterId,
      lesson_id: lessonId,
      note_id: id
    };

    dispatch(
      coursesActions.deleteNotes({
        data: payload,
        callback: () =>
          dispatch(
            coursesActions.fetchCourseNotes({
              data: lesson?.course_id
            })
          )
      })
    );
  };

  return {
    deleteNote,
    editNote
  };
};
