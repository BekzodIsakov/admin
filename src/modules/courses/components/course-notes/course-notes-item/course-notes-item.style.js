import styled from 'styled-components';

export const CourseNotesItemWrapper = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: space-between;

  &:not(:last-child) {
    margin-bottom: 20px;
  }
`;

export const CourseNotesItemTime = styled.h4`
  background: rgba(186, 141, 91, 0.1);
  border-radius: 100px;
  padding: 4px 8px;
  font-style: normal;
  font-weight: 600;
  font-size: 10px;
  line-height: 14px;
  text-align: center;
  margin-right: 10px;
  letter-spacing: -0.02em;
  color: #ba8d5b;
  cursor: pointer;

  @media only screen and (max-width: 1024px) {
    margin-right: 8px;
  }
`;

export const CourseNotesCenter = styled.div`
  width: 100%;
`;

export const CourseNotesCenterTop = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  div:first-child {
    display: flex;

    @media only screen and (max-width: 1024px) {
      flex-direction: column;
    }
  }

  @media only screen and (max-width: 1024px) {
    align-items: flex-start;
  }
`;

export const CourseNotesItemTitle = styled.h3`
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  letter-spacing: -0.02em;
  color: #000000;
  margin-right: 8px;

  @media only screen and (max-width: 1024px) {
    font-size: 13px;
    margin-bottom: 2px;
  }
`;

export const CourseNotesItemSubtitle = styled.h4`
  font-style: normal;
  font-weight: 600;
  font-size: 12px;
  line-height: 16px;
  letter-spacing: -0.02em;
  color: rgba(0, 0, 0, 0.5);

  @media only screen and (max-width: 1024px) {
    font-size: 10px;
    margin-bottom: 8px;
  }
`;

export const CourseNotesItemText = styled.p`
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 22px;
  letter-spacing: -0.02em;
  color: rgba(0, 0, 0, 0.5);
  max-width: 700px;
`;

export const CourseNotesItemRight = styled.div`
  display: flex;
  align-items: center;
`;

export const CourseNotesItemBtn = styled.button`
  background: transparent;
  border: none;
  outline: none;
`;
