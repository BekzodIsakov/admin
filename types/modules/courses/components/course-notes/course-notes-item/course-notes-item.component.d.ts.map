{"version":3,"file":"course-notes-item.component.d.ts","sourceRoot":"","sources":["../../../../../../src/modules/courses/components/course-notes/course-notes-item/course-notes-item.component.js"],"names":[],"mappings":"AAiBA,kCA2BE"}