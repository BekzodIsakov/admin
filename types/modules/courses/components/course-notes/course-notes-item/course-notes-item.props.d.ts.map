{"version":3,"file":"course-notes-item.props.d.ts","sourceRoot":"","sources":["../../../../../../src/modules/courses/components/course-notes/course-notes-item/course-notes-item.props.js"],"names":[],"mappings":"AAGO;;;;;;EAyCN"}