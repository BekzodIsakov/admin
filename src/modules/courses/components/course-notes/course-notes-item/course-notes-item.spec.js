import { customRender } from '@test-utils/helpers';
import { screen, fireEvent } from '@testing-library/react';

import { CourseNotesItem } from './course-notes-item.component';

const setup = ({
  time,
  title,
  subtitle,
  text,
  id,
  deleteNote,
  editNote,
  onTimeClick
}) => {
  return customRender(
    <CourseNotesItem
      time={time}
      title={title}
      subtitle={subtitle}
      text={text}
      id={id}
      deleteNote={deleteNote}
      editNote={editNote}
      onTimeClick={onTimeClick}
    />
  );
};

const allFn = jest.fn();

test('course notes items', () => {
  const { asFragment } = setup({
    time: 'time',
    title: 'title',
    subtitle: 'subtitle',
    text: 'text',
    id: '1',
    deleteNote: allFn,
    editNote: allFn,
    onTimeClick: allFn
  });
  expect(asFragment).toMatchSnapshot();
});

const TEST_MESSAGE = 'Test message';
const TESTID = '1';
const EVENT = { target: { value: TEST_MESSAGE } };

test('Course notes test props', () => {
  setup({
    time: 'time',
    title: 'title',
    subtitle: 'subtitle',
    text: 'text',
    id: TESTID,
    deleteNote: allFn,
    editNote: allFn,
    onTimeClick: allFn
  });
  const courseNotesTitle = screen.getByRole('heading', { name: 'title' });
  const courseNotesText = screen.getByTestId('course-notes-text');
  const courseNotesSubtitle = screen.getByTestId('course-notes-subtitle');
  expect(
    courseNotesTitle && courseNotesText && courseNotesSubtitle
  ).toBeTruthy();
});

test('Course notes edit btn test test', () => {
  setup({
    time: 'time',
    title: 'title',
    subtitle: 'subtitle',
    text: 'text',
    id: TESTID,
    editNote: allFn,
    onTimeClick: allFn,
    deleteNote: allFn
  });
  const courseNotesEditBtn = screen.getByTestId(`course-notes-edit-${TESTID}`);
  fireEvent.click(courseNotesEditBtn, EVENT);
  expect(allFn).toReturn();
});

test('Course notes test delete btn', () => {
  setup({ deleteNote: allFn, editNote: allFn });
  const courseNotesDeleteBtn = screen.getByTestId('course-notes-delete');
  fireEvent.click(courseNotesDeleteBtn, EVENT);
  expect(allFn).toReturn();
});

test('Course notest test time btn', () => {
  setup({ editNote: allFn, onTimeClick: allFn, deleteNote: allFn });
  const courseTimeBtn = screen.getByTestId('course-notes-timeBtn');
  fireEvent.click(courseTimeBtn);
  expect(allFn).toReturn();
});
