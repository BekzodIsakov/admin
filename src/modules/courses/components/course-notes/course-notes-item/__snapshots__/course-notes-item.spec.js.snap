// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`course notes items 1`] = `[Function]`;
